src/internal_rules.jl

```diff
@@ -557,3 +557,132 @@ function EnzymeRules.reverse(
+    if RT <: Const
+        return fact
+    end
+    # TODO: This will be a problem for sparse matrices as invL and dL are dense
+    invL = inv(fact.L)
+    # TODO: dL is dense even when L was sparse
+    dL = Matrix(fact.L * LowerTriangular(invL * A.dval * invL' * 0.5 * I))
+    dfact = Cholesky(dL, 'L', 0)
+    if RT <: DuplicatedNoNeed
+        return dfact
+    else
+        return Duplicated(fact, dfact)
+    end
+end
+
+function EnzymeRules.forward(
+        ::Const{typeof(\)},
+        RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated}},
+        fact::Union{Const, Duplicated}, B::Union{Const, Duplicated};
+        kwargs...
+)
+    retval = copy(B.val)
+    ldiv!(fact.val, retval)
+    retdval = []
+    if isa(fact, Duplicated) && isa(B, Duplicated)
+        retdval = similar(retval)
+        mul!(retdval, fact.dval.U, retval)
+        mul!(retdval, fact.dval.L, retdval)
+        retdval .= B.dval .- retdval
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Duplicated) && isa(B, Const)
+        retdval = similar(retval)
+        mul!(retdval, A, B.val)
+        mul!(retdval, -1, retdval)
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Const) && isa(B, Duplicated)
+        retdval = copy(B.dval)
+        ldiv!(fact.val, retdval)
+    elseif isa(fact, Const) && isa(B, Const)
+        nothing
+    else
+        error("Error in forward \\ Enzyme rule $(typeof(fact)) $(typeof(x)).")
+    end
+    if RT <: Const
+        return retval
+    elseif RT <: DuplicatedNoNeed
+        return retdval
+    else
+        return Duplicated(retval, retdval)
+    end
+end
+
+function EnzymeRules.augmented_primal(
+    config,
+    func::Const{typeof(cholesky)},
+    ::Type{RT}, A, kwargs...) where RT
+
+    primal = func.val(A.val; kwargs...)
+    
+    shadow = ntuple(Val(EnzymeRules.width(config))) do _
+        Base.@_inline_meta
+        Enzyme.make_zero(primal,
+			 #=copy_if_inactive=#Val(true))
+    end
+
+    primal = if EnzymeRules.needs_primal(config)
+        primal
+    else
+        nothing
+    end
+
+    @assert !(typeof(A) <: Active)
+
+    if EnzymeRules.width(config) == 1
+	shadow = shadow[1]
+    end
+
+    cache_A = if EnzymeRules.overwritten(config)[2]
+        copy(A.val)
+    else
+        nothing,
+    end
+    return EnzymeRules.AugmentedReturn(primal, shadow, (cacheA, shadow))
+end
+
+function EnzymeRules.reverse(
+    config,
+    ::Const{typeof(cholesky)},
+    ::Type{RT},
+    cache,
+    A;
+    kwargs...
+) where RT
+    cache_A, shadows = cache
+    if !EnzymeRules.overwritten(config)[2]
+	cache_A = A.val
+    end
+    if EnzymeRules.width(config) == 1
+	shadows = (shadows,)
+    end
+    dAs = if EnzymeRules.width(config) == 1
+	(A.dval,)
+    else
+	A.dval
+    end
+
+    fact = func.val(cache_A; kwargs...)
+
+    for (dA, dfact) in zip(dAs, shadows)
+	    println("Custom Cholesky reverse rule")
+	    #  michel whatever is the rule here
+	    # either way it should read from dfact
```

Comment:
@michel2323 the rule should go here.

Reply:
I left your commit in here. Should I change the rule somehow based on this commit?